### DISCUSSION TITLE
A Dualidade do Generalista

### DISCUSSION THREAD

**Discussion by Cleysonlb** [frontendbr/forum#532] [Discussão]
Galera, primeiramente iai beleza?!
Então, estou vivendo um dilema profissional, hoje trabalho em uma empresa diretamente como     "full stack" atuando tanto no front quanto no back (ruby on rails), acho mt legal, esta sendo um puta aprendizado!
Mas... o Design e UX sempre me chamou mais atenção, porém, entretanto, nunca atuei de fato na area, então, fico no dilema se foco inteiramente no que estou trabalhando agora ou se vou me aperfeiçoando para no futuro atuar no Design/UX.
Vocês vivem algo parecido?! Me contem suas experiências!

**Comment by kvnol** [comment]
@Cleysonlb não vivo essa dilema, não sou full-stack, mas amo todas as partes que envolvem o desenvolvimento de uma aplicação (desde o wireframe, até o _push to server_) e acho que posso dar algumas dicas.

Para que você não se enrole com seu emprego atual, continue trabalhando normalmente pois aprendizado sempre é válido, mas busque fazer projetos simples de UX/UI e jogue no seu Behance/Dribbble/GitHub mesmo. Pegue interfaces e recrie elas, desenhe wireframes e tente tirá-los do papel, repense ou pense em arquiteturas de informação antes de sair codando.

Vejo que o UX (a @talitapagani e a @mjcoffeeholick podem dizer melhor que eu) é algo bem mais filosófico e conceitual, antes de ir para o papel ou tela de uma vez: pensar em quem vai usar, como vai ser usado e de que forma será usado. Pensando no público (personas), na arquitetura das informações e na usabilidade.

Teste isso tudo em pequenos projetos como seu blog pessoal, depois (como já disse) tente recriar interfaces que não são muito boas ou que são boas mas que podem melhorar.

Sei que é um nível bem avançado, mas se liga na qualidade do redesign do NYTimes (https://brasil.uxdesign.cc/interfaces-que-eu-queria-ter-feito-nytimes-design-concept-7d963df32ba0)

Enfim, se eu falei besteira, vamos aprender todos juntos! hahaha :metal:

**Comment by fititnt** [comment]
@Cleysonlb sou full stack e comentei um pouco sobre isso em #506.

---

Comentários extras para você:

Olha, eu já incentivei muita gente a crescer rápido e de ultimamente acabei vendo mais sobre metacognição e a forma como as pessoas aprendem e, em vez de ficar dizendo o que é certo e errado, estou vendo que é bem mais interessante procurar algo mais individualizado e focado em o que vai fazer a pessoa usar de fato.

E, para isso, tem forte relação com felicidade e auto realização pessoal, pois ao fazer algo que você não apenas trabalha, mas gosta, você entra mais frequentemente em um estado chamado de flow (fluxo). 

Uma palestra bacana sobre isso em português está aqui https://www.ted.com/talks/mihaly_csikszentmihalyi_on_flow?language=pt-br
